app/controllers/flood_risk_engine/enrollments/steps_controller.rb

```diff
@@ -108,8 +110,13 @@ def enrollment
         @enrollment ||= Enrollment.find_by_token!(params[:enrollment_id])
       end
 
-      def step_not_found
-        Rails.logger.info "Step Mismatch: :#{step} requested when enrollment at :#{enrollment.step}"
+      def step_not_valid(exception)
+        Rails.logger.info "STEP NOT VALID: #{exception.message}"
+        redirect_to error_path(:step_not_valid)
+      end
+
+      def step_not_found(exception)
+        Rails.logger.info "STEP NOT FOUND: #{exception.message}"
         redirect_to step_url
```

Comment:
I've left `step_not_found` so that it works as the "pre-errors-controller" system did. That it, it just logs the issue and redirects to the current step. I've left it this way so that it can be compared to the new - "show an error message and let the user decide" way of doing things - so that we can compare and contrast in demos.